### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by kawaraa**
Week2 homework.

### CODE REVIEW

File: homework-submission/react-week3/src/App.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+  removeItem(item) {
+    this.setState({ todos: this.state.todos.filter(el => el.id !== item.id) });
+  }
+  editMode(item) {
+    this.setState({ editingMode: item.id });
+  }
+  cancelEditMode() {
+    this.setState({ editingMode: false });
+  }
```
Comment (jalal75) [inline homework-submission/react-week3/src/App.js]: it is not good to create a variable and then change the type of it even if the code is working
( it is boolean and then changed to number)
and you did not give a default value to this variable in the state

File: homework-submission/react-week3/src/App.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  addItem(e) {
+    e.preventDefault();
+    this.state.todos.push({
+      id: Math.max(...this.state.todos.map(el => el.id)) + 1,
+      description: e.target.description.value,
+      deadline: e.target.deadline.value,
+      done: false
+    });
```
Comment (jalal75) [inline homework-submission/react-week3/src/App.js]: not good to work directly in the data
batter to take a copy like
todos = [... this.state.todos]

File: homework-submission/react-week3/src/App.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    this.setState({ todos: this.state.todos.filter(el => el.id !== item.id) });
+  }
+  editMode(item) {
+    this.setState({ editingMode: item.id });
+  }
+  cancelEditMode() {
+    this.setState({ editingMode: false });
+  }
+  updateItem(item) {
+    this.state.todos.forEach(el => {
```
Comment (jalal75) [inline homework-submission/react-week3/src/App.js]: costly code with loop, try to find another way
also working directly with the data

File: homework-submission/react-week3/src/App.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      if (el.id === item.id) {
+        el.description = item.description;
+        el.deadline = item.deadline;
+      }
+    });
+    this.setState({ editingMode: false, todos: this.state.todos });
+  }
+  markAs(item) {
+    let Item = this.state.todos[this.state.todos.indexOf(item)];
```
Comment (jalal75) [inline homework-submission/react-week3/src/App.js]: costly code with indexOf ( it is also a loop ), try to find another way

File: homework-submission/react-week3/src/App.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      }
+    });
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <AddingTodoForm addItem={this.addItem} />
+        <ul className="list">
+          {!this.state.todos[0] ? <h2>No items...</h2> : this.renderItems()}
```
Comment (jalal75) [inline homework-submission/react-week3/src/App.js]: better to be :
this.state.todos.length === 0

File: homework-submission/react-week3/src/render-todo.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+export default function RenderTodo(props) {
+  let styles = { textDecoration: props.todo.done ? "line-through" : "" }
```
Comment (jalal75) [inline homework-submission/react-week3/src/render-todo.js]: don't make any style in Js file
give it an className and identified  it in css file

**Comment by kawaraa** [comment]
Thanks for your comment! I will implement it as recommended.

**Comment by kawaraa** [comment]
It's done you can check it out!

**Pull request closed without merging**